ferrocene/tools/self-test/src/binaries.rs

```diff
@@ -180,7 +198,37 @@ mod tests {
     fn test_check_binary_file_is_a_directory() {
         let utils = TestUtils::new();
         std::fs::create_dir_all(utils.sysroot().join("bin").join("rustc")).unwrap();
-        assert_not_found(utils);
+        match check_binary(utils.reporter(), utils.sysroot(), "rustc", CommitHashOf::Rust) {
+            Ok(()) => panic!("should've failed"),
+            Err(Error::MissingBinary { directory, name, kind }) => {
+                assert_eq!(utils.sysroot().join("bin"), directory);
+                assert_eq!("rustc", name);
+                assert_eq!(MissingBinaryErrorKind::FoundDirectory, kind);
+            }
+            Err(err) => panic!("unexpected error: {err}"),
+        }
+    }
+
+    #[test]
+    fn test_check_binary_file_is_a_symlink() {
+        let utils = TestUtils::new();
+        let original = utils.bin("rustc").create();
+        let name = "link";
+        let link = original.parent().unwrap().join(name);
+
+        #[cfg(unix)]
+        std::os::unix::fs::symlink(&original, &link).unwrap();
+        #[cfg(windows)]
+        std::os::windows::fs::symlink_file(&original, &link).unwrap();
```

Comment:
This means the test suite will break if the user is on a Windows system without symlinks enabled.